src/test/java/org/takes/http/MkSocket.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.http;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.Socket;
+
+/**
+ * Socket mock for reuse.
+ *
+ * @author Nicolas Filotto (nicolas.filotto@gmail.com)
+ * @version $Id$
+ * @since 0.32
+ */
+public final class MkSocket  extends Socket {
+
+    /**
+     * The address to provide for testing purpose.
+     */
+    private final transient InetAddress address;
+
+    /**
+     * The output stream of the socket.
+     */
+    private final transient ByteArrayOutputStream output;
+
+    /**
+     * The input stream of the socket.
+     */
+    private final transient InputStream input;
+
+    /**
+     * Constructs a {@code MkSocket} with the specified input stream.
+     * @param input The input stream of the socket.
+     * @throws IOException in case the call to
+     *  {@link InetAddress#getLocalHost()} fails.
+     */
+    public MkSocket(final InputStream input) throws IOException {
+        super();
+        this.address = InetAddress.getLocalHost();
```

Comment:
@essobedo use `InetAddress.getLoopbackAddress()` and remove `throws IOException`